public/app/features/search/components/DashboardListPage.tsx

```diff
@@ -17,25 +17,24 @@ interface Props {
 
 export const DashboardListPage: FC<Props> = memo(({ navModel, uid }) => {
   const { loading, value } = useAsync(() => {
-    if (uid) {
-      return backendSrv.getFolderByUid(uid).then((folder: any) => {
-        const url = locationUtil.stripBaseFromUrl(folder.url);
+    if (!uid) {
+      return Promise.resolve({ pageNavModel: navModel });
+    }
+    return loadFolderPage(uid, 'manage-folder-dashboards').then(({ folder, model }) => {
+      const url = locationUtil.stripBaseFromUrl(folder.url);
 
-        if (url !== location.pathname) {
-          getLocationSrv().update({ path: url });
-        }
+      if (url !== location.pathname) {
+        getLocationSrv().update({ path: url });
+      }
 
-        return folder.id;
-      });
-    } else {
-      return Promise.resolve(undefined);
-    }
+      return { id: folder.id, pageNavModel: { ...navModel, ...model } };
```

Comment:
Merge with the `navModel`, since the loaded model is missing `node` prop, not sure if necessary or can just be let empty.